pages/denuncias/index.jsx

```diff
@@ -13,17 +13,89 @@ import Layout from '../../components/complaints/Layout'
 import ComplaintCardContent from '../../components/complaints/ComplaintCardContent'
 import { mapComplaint } from '../../utils/complaints/mapper'
 import ComplaintDialog from '../../components/complaints/ComplaintDialog'
+import { format } from 'date-fns'
 
 export default function ComplaintsHomePage () {
   const { data, error, mutate } = useSWR('/api/complaints', (url) => axios.get(url).then(res => res.data))
   const { isOpen, onOpen, onClose } = useDisclosure()
   const [selectedData, setSelectedData] = useState()
+  const [enteredFromDate, setEnteredFromDate] = useState('')
+  const [enteredToDate, setEnteredToDate] = useState('')
+  const currentDate = format(new Date(), 'yyyy-MM-dd')
+
+  const filterByRangeDate = (data, fromDate, toDate) => {
+    // eslint-disable-next-line array-callback-return
+    return data.filter(item => {
+      if (fromDate && !toDate) {
+        // Si desde tiene y hasta no, entonces se filtra desde la fecha desde hasta la fecha actual
+
+        // console.log('desde - 0')
+
+        // item
+        // 01 - 10 - 2022
+        // 02 - 10 - 2022
+        // 03 - 10 - 2022
+        // 04 - 10 - 2022
+        // 12 - 10 - 2022
+
+        // ejemplo
+        // entered from Date -> 03 - 10 - 2022
+        // current date -> 12 - 10 - 2022
+
+        return new Date(fromDate) <= new Date(item['Fecha de denuncia']) && new Date(currentDate) >= new Date(item['Fecha de denuncia'])
```

Comment:
Use date-fns methods to compare.

https://date-fns.org/v2.29.3/docs/isBefore
https://date-fns.org/v2.29.3/docs/isAfter
https://date-fns.org/v2.29.3/docs/isEqual